src/sentry/uptime/detectors/tasks.py

```diff
@@ -197,8 +206,13 @@ def get_failed_url_key(url: str) -> str:
 
 
 def check_url_robots_txt(url: str) -> bool:
-    # TODO: Implement this check
-    return True
+    return get_robots_txt_parser(url).can_fetch("*", url)
```

Comment:
We probably need to use a specific user agent here, maybe that should be a constant or something.

Will this attempt to hit an external url during tests?

Reply:
That's true, would we just want to match the user agent here with whatever the user agent is for when we make the requests on the checker side?

Also, it won't call the external url, I override the `get_robots_txt_parser` function with a mock which returns a RobotFileParser that I manually pass a set of robot.txt lines to parse. Not sure if thats the best way to go about this, but it already feels a bit weird since I'm testing the functionality of the library a little bit here now